timeseries/src/autogluon/timeseries/models/chronos/utils.py

```diff
@@ -55,6 +221,38 @@ def __iter__(self):
             self.callback()
 
 
+class EvaluateAndSaveFinalStepCallback(TrainerCallback):
+    """Callback evaluate and save the model at last training step."""
+
+    def on_step_end(self, args, state, control, **kwargs):
+        if state.global_step >= state.max_steps:
+            control.should_log = True
+            control.should_evaluate = True
+            control.should_save = True
+
+
+class TimeLimitCallback(TrainerCallback):
+    def __init__(self, time_limit: int):
+        """
+        Callback to stop training once a specified time has elapsed.
+
+        Parameters
+        ----------
+        time_limit: int
+            maximum time allowed for training in seconds.
+        """
+        self.time_limit = time_limit
+        self.start_time = None
+
+    def on_train_begin(self, args, state, control, **kwargs):
+        self.start_time = time.time()
```

Comment:
Nit: `time.monotonic()` is a better alternative that avoids some edge cases for `time.time()` (e.g., if system clock changes during training)

Reply:
Fixed.